### DISCUSSION TITLE
Disable SPA behaviour

### DISCUSSION THREAD

**Discussion by Rodeoclash** [vuepress/core#1434] [Q&A]
I've implemented a Cloudflare Worker to provide basic auth protection to a few routes in my docs, however the SPA behaviour of the application will bypass this because clicking on a link doesn't send a request to the server.

Do I have any straightforward options to disable the SPA behaviour vuepress-next? I'm happy with a quick hack to simply make every request route via the server.

I've experimented with both `shouldPreload` and `shouldPrefetch`, but my understanding of these is that they mainly affect the way `<a>` tags get various attributes on them.

**Comment by Mister-Hope** [comment]
TBH, VuePress isn't designed for this. So there won't be a pleasant way doing this.

**Comment by meteorlxy** [comment]
Maybe you can try to replace all `<RouterLink>` with `<a>`

**Comment by Rodeoclash** [comment]
Thanks for the suggestion. In the end I switched to VitePress and used https://vitepress.dev/guide/mpa-mode
